processing/src/main/java/org/apache/druid/query/UnsupportedQueryException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query;
+
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class UnsupportedQueryException extends QueryException
+{
+  private static final String ERROR_CLASS = UnsupportedQueryException.class.getName();
+  public static final String ERROR_CODE = "Unsupported query";
+
+  public UnsupportedQueryException(String errorMessage)
```

Comment:
nit:since this error message is user facing - do you think we should modify the constructor take in an error code, and generate the message in this class?

That would make it less error prone for another dev to pass in sensitive information as part of the error message.

This suggestion might be over-engineering. Feel free to ignore.

Reply:
I feel like keeping an error code to error message mapping will be a pain. There is not a well define set of error code. We can have an unbound number of query type / query feature we don't support. This list would always go up (as we add more functionality) and down (as we fix / iterate). The errorMessage should just indicate what feature exactly is not supported.